src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -112,9 +112,19 @@ public int Count
         
         public override int GetHashCode()
         {
-            return null == _array
-                        ? 0
-                        : _array.GetHashCode() ^ _offset ^ _count;
+            if (_array == null)
+            {
+                return 0;
+            }
+            
+            int hash = 5381;
+            hash = System.Numerics.Hashing.HashHelpers.Combine(hash, _offset);
+            hash = System.Numerics.Hashing.HashHelpers.Combine(hash, _count);
+            
+            // The array hash is expected to be an evenly-distributed mixture of bits,
+            // so rather than adding the cost of another rotation we just xor it.
+            hash ^= _array.GetHashCode();
+            return hash;
```

Comment:
Removed the `FEATURE_CORECLR` ifdefs, @jkotas you mentioned there was a change in how PRs are ported to desktop @ https://github.com/dotnet/coreclr/pull/6767#issuecomment-254331763